docs/language/ql-handbook/types.rst

```diff
@@ -482,35 +482,35 @@ program, so it's helpful to extend a new type (namely ``TTaintType``)::
     
 .. _type-unions:
 
-Type Unions
+Type unions
 ***********
 
 .. note::
     The syntax for type unions is considered experimental and is subject to change.
-    However, they appear in the `standard QL libraries <https://github.com/github/codeql>`.
-    The following sections should help you understand those examples
+    However, type unions appear in the `standard QL libraries <https://github.com/github/codeql>`__.
+    The following sections should help you understand those examples.
 
 Type unions are user-defined types that are declared with the keyword ``class``.
-The syntax resembles type aliases, but with two or more type expressions on the right-hand side.
+The syntax resembles :ref:`type aliases <type-aliases>`, but with two or more type expressions on the right-hand side.
 
-Type unions are used for creating restricted versions of existing algebraic datatypes, by explicitly
-selecting a subset of the branches of said datatype and binding them to a new type.
-In addition to this, type unions of database types are also supported.
+Type unions are used for creating restricted versions of existing :ref:`algebraic datatypes <algebraic-datatypes>`, by explicitly
+selecting a subset of the branches of those datatypes and binding them to a new type.
```

Comment:
Can we make explicit that the branches that are selected for one given type union have to belong to the same datatype here? That might not be entirely clear here.

Perhaps it would already be sufficient to say "selecting subsets of the branches", i.e. making subsets plural there.

Reply:
alternatively "selecting a subset of the branches of a datatype and ..."

Reply:
Ah, thanks for clarifying. How about "Type unions are used for creating restricted versions of an existing algebraic datatype, by explicitly selecting a subset of the branches of that datatype ... "